File: algorithm_study_python/baekjoon_online/p_1934/p_1934_2.py
'''
input:
10
1 45000
6 10
13 17
180 200
18 20
10 20
1679 874
3762 4047
6080 3876
78963 23075

output:
45000
30
221
1800
180
20
63802(GCD:23)
267102(GCD:57)
310080(GCD:76)
1822071225(GDC:1)

'''


def find_greatest_common_divisor(a, b):
    x = max(a, b)
    y = min(a, b)
    while (x > 0) & (y > 0):
        temp = y
        y = x % y
        x = temp

    return x

def find_least_common_multiple(a, b):
    return int((a * b) / find_greatest_common_divisor(a, b))


# Main
line_count = int(input())
results = []

for i in range(line_count):
    a, b = input().split(" ")
    a = int(a)
    b = int(b)

    result = find_least_common_multiple(a, b)

    results.append(result)

for i in results:
    print(i)

File: algorithm_study_java/src/baekjoon_online/p_1934/p_1934_2.java
package baekjoon_online.p_1934;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

/*
Input
3
1 45000
6 10
13 17

Output
45000
30
221
 */
public class p_1934_2 {
    public static void main(String[] args) {
        // First line
        BufferedReader inputReader = new BufferedReader(new InputStreamReader(System.in));
        String line = null;
        try {
            line = inputReader.readLine();
        } catch (IOException e) {
            e.printStackTrace();
        }

        int count = Integer.parseInt(line.trim());
        while (count-- > 0) {
            try {
                line = inputReader.readLine();
            } catch (IOException e) {
                e.printStackTrace();
            }
            String[] abStrings = line.trim().split(" ");
            int a = Integer.parseInt(abStrings[0].trim());
            int b = Integer.parseInt(abStrings[1].trim());

            System.out.println(a * b / gcd(a, b));
        }
    }

    private static int gcd(int a, int b) {
        int max = a;
        int min = b;
        do {
            int temp = max % min;
            max = min;
            min = temp;
        } while (min != 0);

        return max;
    }
}
